neon_utils/packaging_utils.py

```diff
@@ -265,3 +266,26 @@ def get_skill_license():  # TODO: Implement OSM version of this
     skill_data = dict_merge(default_skill, skill_data)
     skill_data["requirements"]["python"].sort()
     return dict(dict_merge(skill_data, readme_data))
+
+def install_packages_from_pip(module: str, packages: List[str]) -> int:
+    """
+    Install a Python package using pip
+    :param module: string neon module to install dependency for
+    :param packages: List(string) list of packages to install
+    :returns: int pip exit code
+    """
+    import pip
+    _, tmp_file = mkstemp()
+
+    install_str = " ".join(packages)
+    with open(tmp_file, 'w', encoding="utf8") as f:
+        constraints = '\n'.join(get_package_dependencies(module))
+        f.write(constraints)
+        LOG.info(f"Constraints={constraints}")
+    LOG.info(f"Requested installation of plugins: {install_str}")
+    for pkg in packages:
```

Comment:
If passing a string doesn't work, consider writing to a temp file and installing with `-r`. Installing one at a time at best will add some overhead to resolving compatible versions and at worst could cause a specified package to break or fail because a dependency was overridden by a later package in the list